source/Coop.Core/Client/Services/Clans/Handler/ClientClanHandler.cs

```diff
@@ -0,0 +1,206 @@
+﻿using Common.Logging;
+using Common.Messaging;
+using Common.Network;
+using GameInterface.Services.ObjectManager;
+using Serilog;
+using GameInterface.Services.Clans.Messages;
+using Coop.Core.Client.Services.Clans.Messages;
+using Coop.Core.Server.Services.Clans.Messages;
+using System;
+
+namespace Coop.Core.Client.Services.Clans.Handler
+{
+    /// <summary>
+    /// Handles all changes to clans.
+    /// </summary>
+    public class ClientClanHandler : IHandler
+    {
+        private readonly IMessageBroker messageBroker;
+        private readonly INetwork network;
+        private readonly ILogger Logger = LogManager.GetLogger<ClientClanHandler>();
+
+        public ClientClanHandler(IMessageBroker messageBroker, INetwork network)
+        {
+            this.messageBroker = messageBroker;
+            this.network = network;
+            messageBroker.Subscribe<ClanNameChange>(Handle);
+            messageBroker.Subscribe<NetworkClanNameChangeApproved>(Handle);
+            
+
+            messageBroker.Subscribe<ClanLeaderChange>(Handle);
+            messageBroker.Subscribe<NetworkClanLeaderChangeApproved>(Handle);
+
+            messageBroker.Subscribe<ClanKingdomChange>(Handle);
+            messageBroker.Subscribe<NetworkClanKingdomChangeApproved>(Handle);
+
+            messageBroker.Subscribe<DestroyClan>(Handle);
+            messageBroker.Subscribe<NetworkDestroyClanApproved>(Handle);
+
+            messageBroker.Subscribe<AddCompanion>(Handle);
+            messageBroker.Subscribe<NetworkCompanionAddApproved>(Handle);
+
+            messageBroker.Subscribe<AddRenown>(Handle);
+            messageBroker.Subscribe<NetworkRenownAddApproved>(Handle);
+
+            messageBroker.Subscribe<ChangeClanInfluence>(Handle);
+            messageBroker.Subscribe<NetworkClanChangeInfluenceApproved>(Handle);
+
+            messageBroker.Subscribe<AdoptHero>(Handle);
+            messageBroker.Subscribe<NetworkAdoptHeroApproved>(Handle);
+
+            messageBroker.Subscribe<LocalNewHeir>(Handle);
+            messageBroker.Subscribe<NetworkNewHeirApproved>(Handle);
```

Comment:
I highly recommend separating these into different handler files under the `Clans` service